src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -402,30 +402,33 @@ function VisEditor(
     $scope.$listenAndDigestAsync(timefilter, 'timeUpdate', updateTimeRange);
     $scope.$listenAndDigestAsync(timefilter, 'refreshIntervalUpdate', updateRefreshInterval);
 
-    // update the searchSource when filters update
-    const filterUpdateSubscription = subscribeWithScope($scope, queryFilter.getUpdates$(), {
-      next: () => {
-        $scope.filters = queryFilter.getFilters();
-        $scope.fetch();
-      }
-    });
-
     // update the searchSource when query updates
     $scope.fetch = function () {
       $state.save();
       savedVis.searchSource.setField('query', $state.query);
       savedVis.searchSource.setField('filter', $state.filters);
-      $scope.globalFilters = queryFilter.getGlobalFilters();
       $scope.vis.forceReload();
     };
 
+    // update the searchSource when filters update
+    const filterUpdateSubscription = subscribeWithScope($scope, queryFilter.getUpdates$(), {
+      next: () => {
+        $scope.filters = queryFilter.getFilters();
```

Comment:
Did you want all filters or only the non-global filters?
If you want the non-global filters - use `getAppFilters`

Reply:
No, we want all filters here, since it's what's passed to the filter bar.